kafka-client/src/main/java/dev/responsive/kafka/store/ResponsiveStore.java

```diff
@@ -100,176 +73,110 @@ public void init(final ProcessorContext context, final StateStore root) {
 
   @Override
   public void init(final StateStoreContext context, final StateStore root) {
-    try {
-      LOG.info("Initializing state store {}", name);
-      StoreUtil.validateTopologyOptimizationConfig(context.appConfigs());
-      this.context = asInternalProcessorContext(context);
-
-      partition = context.taskId().partition();
-      client.createDataTable(name.cassandraName());
-      initRemote.await(Duration.ofSeconds(60));
-      LOG.info("Remote table {} is available for querying.", name.cassandraName());
-
-      client.prepareStatements(name.cassandraName());
-      client.initializeOffset(name.cassandraName(), partition);
-
-      buffer = new CommitBuffer<>(
-          client,
-          name.cassandraName(),
-          new TopicPartition(
-              changelogFor(context, name.kafkaName(), false),
-              partition),
-          asRecordCollector(context),
-          admin,
-          PLUGIN
-      );
-
-      open = true;
-
-      context.register(root, buffer);
-    } catch (InterruptedException | TimeoutException e) {
-      throw new ProcessorStateException("Failed to initialize store.", e);
+    if (context instanceof GlobalProcessorContextImpl) {
```

Comment:
So slick! Also I like the new names -- I imagine a future where users don't think in terms of Kafka Streams "global" vs "local" stores but rather "full topic" vs "partitioned" views. Ultimately there's just one physical table but your processors can subscribe to a read-only "full view" or a read-write "partition view"

Going even further we can have IQ be more natively and intuitively integrated, like it's just a regular view that's read-only and either "full", "partition", or "single-key" (point lookup)

Maybe that's the real insight here...what is IQ if not an external-facing global table? 😈

Reply:
hah! I didn't even think about it that deeply when I named it. "local" just didn't feel right so I called it `Partitioned` but that's a good point